internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -538,6 +546,18 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, bmh *bmh_v1alpha
 		return reconcileError{err}
 	}
 
+	// Add detached annotation to hub BMH so that Ironic stops managing the host from
+	// the hub cluster. After spoke BMH is created, the host will be managed by the spoke
+	// cluster.
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
```

Comment:
maybe we can use 
host.ObjectMeta.SetAnnotations(annotations)

Reply:
The `annotations` parameter is type `map[string]string`. In our case, we just want to set a single annotation and not the entire map.